core/src/bls_sigverify/bls_sigverifier.rs

```diff
@@ -0,0 +1,1575 @@
+//! The BLS signature verifier.
+
+use {
+    crate::{
+        bls_sigverify::{
+            bls_sigverify_service::BLSSigVerifyServiceError, stats::BLSSigVerifierStats,
+        },
+        cluster_info_vote_listener::VerifiedVoteSender,
+    },
+    agave_votor::consensus_metrics::{ConsensusMetricsEvent, ConsensusMetricsEventSender},
+    agave_votor_messages::{
+        consensus_message::{Certificate, CertificateType, ConsensusMessage, VoteMessage},
+        vote::Vote,
+    },
+    ahash::{AHashMap, AHashSet},
+    bitvec::prelude::{BitVec, Lsb0},
+    crossbeam_channel::{Sender, TrySendError},
+    rayon::iter::{
+        IndexedParallelIterator, IntoParallelIterator, IntoParallelRefIterator, ParallelIterator,
+    },
+    solana_bls_signatures::{
+        pubkey::{Pubkey as BlsPubkey, PubkeyProjective, VerifiablePubkey},
+        signature::SignatureProjective,
+        BlsError,
+    },
+    solana_clock::Slot,
+    solana_measure::measure::Measure,
+    solana_pubkey::Pubkey,
+    solana_runtime::{bank::Bank, bank_forks::SharableBanks, epoch_stakes::BLSPubkeyToRankMap},
+    solana_signer_store::{decode, DecodeError},
+    solana_streamer::packet::PacketBatch,
+    std::{
+        collections::{HashMap, HashSet},
+        sync::{atomic::Ordering, Arc, RwLock},
+        time::Instant,
+    },
+    thiserror::Error,
+};
+
+// TODO(sam): This logic of extracting the message payload for signature verification
+//            is brittle, but another bincode serialization would be wasteful.
+//            Revisit this to figure out the best way to handle this.
+
```

Comment:
Did this comment get disconnected from its code ? It doesn't appear to be relevant to the function immediately below

Reply:
I think this one is done for now, removing.